fix(client): add superBlock check to handle Rosetta being treated as a project block

JordanMooree opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #46006

<!-- Feel free to add any additional description of changes below this line -->
The code adds the following condition:
`challenge.superBlock !== 'coding-interview-prep'` as the superBlock check.

It produces the following result for Rosetta:
![image](https://user-images.githubusercontent.com/27147016/169399693-12d3f960-a030-4ea3-8156-44366232cc11.png)

Hope it resolved the issue!

JordanMooree committed:
add superBlock condition to handle rosetta

gikf reviewed (dismissed):
Looks good :+1:

raisedadead commented on client/src/templates/Introduction/components/block.tsx:
```diff
@@ -122,7 +122,8 @@ export class Block extends Component<BlockProps> {
     const isProjectBlock = challenges.some(({ challenge }) => {
       const isJsProject =
         [3, 6, 10, 14, 17].includes(challenge.order) &&
-        challenge.challengeType === 5;
+        challenge.challengeType === 5 &&
+        challenge.superBlock !== 'coding-interview-prep';
```
🙄

moT01 replied:
We need to rework our challengeTypes - each type should have it's own type. The JS projects are the same type as the rest of the JS challenges, it should be its own.

raisedadead reviewed (commented):
@JordanMooree 

Thanks for the PR and your hard work. 

Everything below this line is not for you directly. It's for us, maintainers.

--

This change should fix the symptom, but trust me, it stinks, to be blunt. Can we make this better @freeCodeCamp/dev-team 

I don't particularly appreciate that we need to use the `isJsProject` as a classifier for this proposed fix.

Also, the whole logic in this component is a bit convoluted.

ojeytonwilliams commented:
We've mixed up concerns. The curriculum should be telling the client what to display and the client should just render it, rather than having to first figure out what it's supposed to be displaying.

nhcarrigan commented:
@ojeytonwilliams realistically, would a better approach be a `isStepBased` key in the block meta? Which we could then use in the client to render it as a grid instead of a list?

ojeytonwilliams commented:
@nhcarrigan yep, that sounds good. As long as the client just has to inspect a single property of a challenge, that's fine.

When we rework things, I think we'll be wanting to send the client block/superBlock info so it knows what to render. Our current approach of only passing challenge nodes and inferring everything from them is rather clunky, to say the least.

nhcarrigan commented:
@JordanMooree our proposed solution is a bit larger in scope. Is this something you'd be interested in giving a shot?

JordanMooree commented:
@nhcarrigan Yes, I'd be interested! Can you/anyone offer guidance on the proposed solution?

JordanMooree committed:
Merge branch 'freeCodeCamp:main' into main

nhcarrigan commented:
So the goal would be to add an `isStepBased: true` property to the `meta.json` files for the blocks that are step-based projects - then update that logic you were working on in `block.tsx` to look for that meta property instead of the logic we currently have.

There might be a couple of other places to update the logic too. If you'd like to give it a shot, you're welcome to push more commits to this branch. You can ping here when you're ready for another review, or ask in our [contributor chat](https://chat.freecodecamp.org/channel/contributors) if you have questions.

JordanMooree committed:
isStepBased added to meta files and ChallengeNode

JordanMooree committed:
Merge branch 'freeCodeCamp:main' into main

JordanMooree commented:
Hey everyone, so I pushed a commit. Let me know what you think and any ideas on what I can do to further clean the code

JordanMooree committed:
Merge branch 'freeCodeCamp:main' into main

renovate[bot] committed:
chore(deps): update babel monorepo (#46095)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

renovate[bot] committed:
chore(deps): update babel monorepo (#46095)

Co-authored-by: Renovate Bot <bot@renovateapp.com>

JordanMooree committed:
fix add stepBased property to in-production blocks

JordanMooree committed:
Merge branch 'freeCodeCamp:main' into main

JordanMooree commented:
Just realized in my last commit I was adding `isStepBased` to blocks that were not in production yet. I fixed it and added the property to [Responsive Web Design](https://www.freecodecamp.org/learn/2022/responsive-web-design/) blocks.

JordanMooree committed:
Update block.tsx

renovate[bot] committed:
chore(deps): update dependency cypress to v9 (#46129)

Co-authored-by: Renovate Bot <bot@renovateapp.com>
Co-authored-by: moT01 <20648924+moT01@users.noreply.github.com>

Dario-DC committed:
fix(curriculum): put correct marker in the hint (#46147)

moT01 committed:
feat: group cypress deps in renovate config (#46150)

Buet23 committed:
fix(curriculum): add switch statement spaces for consistency (#46141)

Co-authored-by: Huyen Nguyen <25715018+huyenltnguyen@users.noreply.github.com>

Sboonny committed:
fix(curriculum): external link in words blank JS (#46145)

moT01 committed:
revert: cypress temporary workaround

reverts #45593

Sboonny committed:
fix(curriculum): external wiki link in JS A&DS (#46144)

* fix(curriculum): external wiki link in JS A&DS

* reorder the phrases

moT01 committed:
Revert "revert: cypress temporary workaround" (#46156)

This reverts commit 9d82633a61ee600f8ce69585f6810a5ba8bd94b7.

Sboonny committed:
fix(curriculum): external floating link in JS (#46159)

Co-authored-by: Sem Bauke <46919888+Sembauke@users.noreply.github.com>

Sboonny committed:
fix(curriculum): external card counting link in JS (#46158)

Sboonny committed:
fix(curriculum): external Mad Libs link in JS steps (#46157)

camperbot committed:
chore(i18n,learn): processed translations (#46149)

Sboonny committed:
fix(curriculum): external passport-GitHub's link in QA (#46116)

Co-authored-by: Lasse Jørgensen <28780271+lasjorg@users.noreply.github.com>

scissorsneedfoodtoo committed:
fix: display certs on profile for username with upper chars (#46118)

Sboonny committed:
fix(curriculum): add hint about using media types (#46055)

Co-authored-by: Huyen Nguyen <25715018+huyenltnguyen@users.noreply.github.com>

lasjorg committed:
fix(product landing page): allow header or children to be fixed (#46139)

Co-authored-by: Tom <20648924+moT01@users.noreply.github.com>

Sboonny committed:
fix(curriculum): removed link in JS basic-JavaScript (#46143)

* fix(curriculum): removed link in JS basic-javascript

* removed JSON and its example

ahmaxed committed:
feat: add a leading submitChallenge debounce (#46085)

* feat: reduce submitChallenge debounce

* feat: register the leading call

* fix: presist debounce calls

* Update client/src/templates/Challenges/classic/editor.tsx

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

moT01 committed:
fix: third-party cookie issues with CodeAlly login (#46090)

camperbot committed:
chore(i18n,learn): processed translations (#46161)

lasjorg committed:
fix: remove parentheses around element type (#46164)

JordanMooree commented:
I will be closing this pull request and taking some time to understand Git and GitHub and reopen another pull request on this issue as soon as I can. I've been completing lost working on this through Git and GitHub 🙃

raisedadead commented:
Hi, @JordanMooree no worries. There is no rush and we have all been there. If you need any help please find us in the Contributors chat room.
